src/store/reducers/network.js

```diff
@@ -237,34 +232,61 @@ const checkNetworkEpic = (providerType, isInitialHealthcheck) => async (dispatch
      */
     dispatch(platformReducer.actions.networkChanged());
     dispatch(platformReducer.actions.web3Reset());
-
     currentNetworkName = getState().getIn(['network', 'activeNetworkName']);
+    dispatch(setTokenAddresses(currentNetworkName));
 
     /**
      * Loading contracts and initializing market
      */
-    try  {
+    try {
       return await Promise.all([
         dispatch(platformReducer.actions.contractsLoaded(contractsBootstrap.init(currentNetworkName))),
         await dispatch(balancesReducer.actions.getDefaultAccountEthBalance()),
         await dispatch(balancesReducer.actions.subscribeAccountEthBalanceChangeEventEpic(window.web3.eth.defaultAccount)),
-        await dispatch(platformReducer.actions.marketInitialized(marketBootstrap.init(dispatch))),
+        await dispatch(platformReducer.actions.marketInitialized(marketBootstrap.init(dispatch, currentNetworkName))),
         dispatch(balancesReducer.actions.getAllTradedTokensBalances(window.contracts.tokens)),
         dispatch(balancesReducer.actions.getAllTradedTokensAllowances(window.contracts.tokens, window.contracts.market.address)),
-        dispatch(balancesReducer.actions.subscribeTokenTransfersEventsEpic(window.contracts.tokens))
       ]).then(
-        () => {
-          dispatch(CheckNetwork.fulfilled());
-        }
+        async () =>
+        {
+          await dispatch(subscribeLatestBlockFilterEpic());
+          const currentLatestBlock = network.latestBlockNumber(getState());
+          /**
+           *  Fetch LogTake events for set historicalRange
+           */
+          dispatch(
```

Comment:
I think this called is duplicated. It is contained in both the `if` and the `else statement

Reply:
ill move this into subroutine.

Reply:
done